config.m4

```diff
@@ -306,7 +306,7 @@ EOT
     if test "$PHP_DDTRACE_SIDECAR_MOCKGEN" != "-"; then
       ddtrace_mockgen_invocation="HOST= TARGET= $PHP_DDTRACE_SIDECAR_MOCKGEN"
     else
-      ddtrace_mockgen_invocation="cd \"$ext_srcdir/components-rs/php_sidecar_mockgen\"; HOST= TARGET= CARGO_TARGET_DIR=\$(builddir)/target/ \$(DDTRACE_CARGO) run"
+      ddtrace_mockgen_invocation="cd \"$ext_srcdir/components-rs/php_sidecar_mockgen\"; HOST=$(rustc -vV | sed -n 's|host: ||p') TARGET=$(rustc -vV | sed -n 's|host: ||p') CARGO_TARGET_DIR=\$(builddir)/target/ \$(DDTRACE_CARGO) run"
```

Comment:
We can't rely on rustc being in PATH (that's also why we do have e.g. `AC_CHECK_TOOL(DDTRACE_CARGO, cargo, [:])` for cargo).

If this is specifically a problem with our Makefile, it must be an interaction with the Makefile. After all, it works when run isolated. I'd rather fix the Makefile than changing config.m4 here.

Reply:
For the context, it seems the `host` & `target` settings require a non-empty string for non-Linux platform.
It works on macOS when run isolated because the `cc` crate is locked to version `1.0.83` where this restriction does not seem to exist.
The error occurs with our Makefile because the latest `cc` crate version is used instead (`1.0.92` at the time of writing).

But to avoid the dependency on the `rustc` binary, I moved the detection to the sidecar mockgen code directly.
WDYT?